### PR TITLE
  CLI | contentstack-cli-content-type plugin required code update

### PR DESCRIPTION
**Pull request by aman19K**
- [x] Oclif core v2 migration changes
- [x] CLI dependency update
- [x] Vulnerability fixes
- [x] Utility plugin implementation 
- [x] Convert HTTP API calls to SDK calls
- [x] Github workflow
- [x] stack & token-alias flag deprecation

Two API implementation pending from sdk side /content_types/${uid}/references & /audit-logs -> using http call for this.

### CODE REVIEW

File: src/commands/content-type/diagram.ts [hunk 1]
```diff
@@ -1,39 +1,50 @@
...
+      parse: printFlagDeprecation(['--token-alias'], ['-a', '--alias'])
+    }),
+
+    alias: flags.string({
+      char: 'a',
+      description: 'Alias of the management token'
     }),
 
     output: flags.string({
       char: 'o',
```
Comment (netrajpatel) [inline src/commands/content-type/diagram.ts:43]: @aman19K Please remove this short flag

File: src/commands/content-type/diagram.ts [hunk 2]
```diff
@@ -1,39 +1,50 @@
...
       description: 'full path to output',
       hidden: false,
       multiple: false,
-      required: true,
+      required: true
     }),
 
     direction: flags.string({
```
Comment (netrajpatel) [inline src/commands/content-type/diagram.ts:50]: @aman19K Please remove this short flag

File: src/commands/content-type/diagram.ts [hunk 3]
```diff
@@ -43,7 +54,7 @@ export default class DiagramCommand extends Command {
       options: ['portrait', 'landscape'],
       hidden: false,
       multiple: false,
-      required: true,
+      required: true
     }),
 
     type: flags.string({
```
Comment (netrajpatel) [inline src/commands/content-type/diagram.ts:60]: @aman19K Please remove this short flag

File: src/commands/content-type/list.ts [hunk 4]
```diff
@@ -1,59 +1,72 @@
...
 import buildOutput from '../../core/content-type/list'
+import { getStack, getContentTypes } from '../../utils'
 
 export default class ListCommand extends Command {
-  static description = 'list all Content Types in a Stack';
+  static description = 'List all Content Types in a Stack'
 
   static examples = [
-    '$ csdx content-type:list -s "xxxxxxxxxxxxxxxxxxx"',
+    '$ csdx content-type:list -k "xxxxxxxxxxxxxxxxxxx"',
```
Comment (netrajpatel) [inline src/commands/content-type/list.ts:10]: @aman19K Use full flag names in examples

File: src/commands/content-type/list.ts [hunk 5]
```diff
@@ -1,59 +1,72 @@
...
+      parse: printFlagDeprecation(['--token-alias'], ['-a', '--alias'])
+    }),
+
+    alias: flags.string({
+      char: 'a',
+      description: 'Alias of the management token'
     }),
 
     order: flags.string({
       char: 'o',
```
Comment (netrajpatel) [inline src/commands/content-type/list.ts:41]: @aman19K Please remove this short flag

**Review by netrajpatel — CHANGES_REQUESTED** [review]

File: .eslintrc [hunk 6]
```diff
@@ -1,11 +1,37 @@
...
+    ],
+    "semi": "off",
+    "indent": "off",
+    "comma-dangle": "off",
+    "unicorn/consistent-function-scoping": "off",
+    "max-params": "off",
+    "no-return-await":"off",
+    "no-process-exit":"off",
+    "no-trailing-spaces":"off",
+    "unicorn/no-process-exit":"off"
```
Comment (antonyagustine) [inline .eslintrc:35]: Why did we disable this many lint checks.?

File: package.json [hunk 7]
```diff
@@ -5,7 +5,8 @@
   "author": "Michael Davis",
   "bugs": "https://github.com/Contentstack-Solutions/contentstack-cli-content-type/issues",
   "dependencies": {
-    "@contentstack/cli-command": "0.1.0-beta",
+    "@contentstack/cli-command": "^1.2.8",
+    "@contentstack/cli-utilities": "^1.4.4",
     "@oclif/command": "^1.8.0",
     "@oclif/config": "^1.17.0",
```
Comment (antonyagustine) [inline package.json:11]: These two packages are no more needed, right? As we should use `@oclif/core`

File: tsconfig.json [hunk 8]
```diff
@@ -8,7 +8,8 @@
     "strict": true,
     "target": "es2017",
     "allowJs": true,
-    "esModuleInterop": true
+    "skipLibCheck": true,
+    "esModuleInterop": true,
```
Comment (antonyagustine) [inline tsconfig.json:12]: Can we also add `"declaration": true` in the `tsconfig.json` and remove the types folder from src.?

File: .eslintrc (hunk 6 above)
Comment (aman19K) [reply .eslintrc]: @antonyagustine reference from contentstack/cli/packages/contentstack-branches/.eslintrc

File: package.json (hunk 7 above)
Comment (aman19K) [reply package.json]: @antonyagustine removed

**Review by netrajpatel — APPROVED** [review]

**Review by antonyagustine — APPROVED** [review]

**Review by aravindbuilt — APPROVED** [review]

**Pull request merged**